addons/mail/static/src/discuss/call/common/call_actions.js

```diff
@@ -191,17 +190,22 @@ export const blurBackgroundAction = {
     sequenceGroup: 200,
 };
 registerCallAction("fullscreen", {
-    btnClass: ({ owner, thread }) =>
+    btnClass: ({ owner, store, thread }) =>
         attClassObjectToString({
             "o-discuss-CallActionList-pulse": Boolean(
-                !owner.env.pipWindow && thread.promoteFullscreen === CALL_PROMOTE_FULLSCREEN.ACTIVE
+                !owner.env.pipWindow &&
+                    !store.rtc.state.isFullscreen &&
+                    !store.rtc.selfSession?.promoteFullscreen &&
+                    thread.videoCountNotSelf > 0 &&
+                    thread.chat_window?.isOpen
             ),
         }),
     condition: ({ store, thread }) => thread?.eq(store.rtc?.channel),
     name: ({ store }) => (store.rtc.state.isFullscreen ? _t("Exit Fullscreen") : _t("Fullscreen")),
     isActive: ({ store }) => store.rtc.state.isFullscreen,
     icon: ({ action }) => (action.isActive ? "fa fa-compress" : "fa fa-expand"),
     onSelected: ({ store }) => {
+        store.rtc.selfSession.promoteFullscreen = true;
         if (store.rtc.state.isFullscreen) {
```

Comment:
That doesn't make sense for clicking on fullscreen to promote fullscreen... I would expect the opposite: it removes the promote fullscreen.